### ISSUE TITLE
Table to support additional elements to make the table readable by a screen reader

### ISSUE THREAD

**Issue by aradhikaIbm** [IBM/carbon-components-angular#189]
## Detailed description

> Describe in detail the issue you're having.
Trying to use the table and set the basic attributes or element (caption) that are needed so a screen reader can read a table. 

Per the accessibility guidelines you need to be able to use the following elements on a table to make it readable:
1. Caption element or Summary attribute
2. Ids on headers so that the td can reference them using "header" attribute
3. Scope attribute to help a identifier the header as a row or col scope.

Here some accessibility links:
https://www.w3.org/WAI/tutorials/tables/
https://www.w3.org/TR/wai-aria-practices-1.1/#table

I've even attached a sample table we create that has 2 headers for a given table cell.

> Is this a feature request (new component, new icon), a bug, or a general issue?
enhancement

> Is this issue related to a specific component?
table

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
IBM Watson Recruitment, Accessibility deadline of Dec 10, 2018.

## Additional information

* Screenshots or code
Code example: 
  <table class="iwrt-table-data" id="iwrt-sucess-tiering-table-_old" summary="summary of our table">
    <thead>
        <tr class="iwrt-table-data-header">
          <th id="col1header" scope="col">xxxxx</th>
          <th id="col2header" scope="col" class="iwrt-numeric-align">xxxxx</th>
          <th id="col5header" scope="col" class="iwrt-numeric-align">xxxxx</th>
          <th id="col6header" scope="col" class="iwrt-numeric-align">xxxxx</th>
          <th id="col7header" scope="col" class="iwrt-numeric-align">xxxxx</th>
        </tr>
    </thead>
    <tbody>
        <tr  class="iwrt-narrow">
          <th id="row1header" class="iwrt-no-wrap" scope="row">row1</th>
          <td headers="row1header col2header" class="iwrt-numeric-align">yyyyyy</td>
          <td headers="row1header col5header" class="iwrt-numeric-align">yyyyyy</td>
          <td headers="row1header col6header" class="iwrt-numeric-align">yyyyyy</td>
          <td headers="row1header col7header" class="iwrt-numeric-align">yyyyyy</td>
        </tr>
        <tr  class="iwrt-narrow">
          <th id="row2header" class="iwrt-no-wrap" scope="row">row2</th>
          <td headers="row2header col2header" class="iwrt-numeric-align">zzzzz</td>
          <td headers="row2header col5header" class="iwrt-numeric-align">zzzzz</td>
          <td headers="row2header col6header" class="iwrt-numeric-align">zzzzz</td>
          <td headers="row2header col7header" class="iwrt-numeric-align">zzzzz</td>
        </tr>
        <tr  class="iwrt-narrow">
          <th id="row3header" class="iwrt-no-wrap" scope="row">row3</th>
          <td headers="row3header col2header" class="iwrt-numeric-align">vvvvv</td>
          <td headers="row3header col5header" class="iwrt-numeric-align">vvvvv</td>
          <td headers="row3header col6header" class="iwrt-numeric-align">vvvvv</td>
          <td headers="row3header col7header" class="iwrt-numeric-align">vvvvv</td>
        </tr>
    </tbody>
  </table>

**Comment by sksiyer** [comment]
Hello, I'd like to fix this issue - could this issue be assigned to me please?

**Comment by cal-smith** [comment]
For sure!

**Comment by cal-smith** [comment]
Thanks for taking this on @sksiyer

**Comment by ninja511** [comment]
Has this been fixed yet? Do you have an estimated timeframe on fix? Thanks

**Comment by sksiyer** [comment]
> @sksiyer Has this been fixed yet? Do you have an estimated timeframe on fix? Thanks

Estimate timeframe for fix Q1 2021? Thank you. Sorry about delay on this
